### PR TITLE
build: add OPAEPlugin.cmake

### PR DESCRIPTION
**Pull request by r-rojo**
* define cmake function called opae_add_shared_plugin that creates a
  shared library that registers a plugin as defined by its PLUGIN
  argument. To make the shared library it:
  * writes a file with C code implementing a library constructor that
    registers a plugin
  * adds any source files as listed after the SOURCE keyword
  * links this to opae-c-archive

The end result is a shared library that implements the OPAE API but has
the named plugin registered when the library loads.

Signed-off-by: Rodrigo Rojo <rodrigo.rojo@intel.com>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -577,5 +577,6 @@ install(FILES
 
 install(FILES
     ${OPAE_SDK_SOURCE}/cmake/modules/OPAEGit.cmake
+    ${OPAE_SDK_SOURCE}/cmake/modules/OPAEPlugin.cmake
```
Comment (tswhison) [inline CMakeLists.txt:580]: Need to add to .rpm specs and .deb install files.

Comment (tswhison) [inline CMakeLists.txt:580]: Need to add to .rpm specs and .deb install files.

Comment (r-rojo) [reply CMakeLists.txt:580]: this should be covered by including the cmake installed directory in lib:
```spec
%{_prefix}/lib/opae-%{version}
```

**Review by tswhison — APPROVED** [review]

**Comment by michael-adler** [comment]
Is this a solution to the libopae-c.so that wraps the ASE library automatically?

**Comment by r-rojo** [comment]
@michael-adler , yes. By linking in libopae-c-archive.a, a library implements the API. This solution also gives a way to "hardcode" a plugin in the library constructor.

**Comment by michael-adler** [comment]
Would it work as a shared library, e.g. libopae-c-ase.so, so that existing scripts continue to work?

**Review by michelleyho — APPROVED** [review]

**Comment by r-rojo** [comment]
yes, the result of using this cmake function is a .so that you can link with to resolve symbols introduced by including <opae/fpga.h>.
I built a simple example that includes opae headers but links with libopae-c-ase. This example only calls fpgaEnumerate but yo can see here that by doing so it reached into ase plugin code.

image

Here is the backtrace:
image

**Review by michael-adler — APPROVED** [review]
Nice!

**Pull request merged**